dalle2_pytorch/train_configs.py

```diff
@@ -16,7 +16,43 @@ def default(val, d):
 def ListOrTuple(inner_type):
     return Union[List[inner_type], Tuple[inner_type]]
 
-# pydantic classes
+# general pydantic classes
+
+class TrainSplitConfig(BaseModel):
+    train: float = 0.75
+    val: float = 0.15
+    test: float = 0.1
+
+    @root_validator
+    def validate_all(cls, fields):
+        actual_sum = sum([*fields.values()])
+        if actual_sum != 1.:
+            raise ValueError(f'{fields.keys()} must sum to 1.0. Found: {actual_sum}')
+        return fields
+
+class TrackerConfig(BaseModel):
```

Comment:
tracker config could assume the responsibility of instantiating either the console or wandb trackers at some point, but could also be another PR

Reply:
Yeah totally agree! Actually, the only reason I didn't do this was because I've been experimenting with Hugging Face Accelerator and they have a built-in WANDB tracker that I've been testing out....not sure I like how restrictive their tracker is, though. (off topic: but hfa has been pretty nice, otherwise, at least so far!)

If you (or someone else) doesn't around to this, perhaps I can just do a seperate PR once I get the distributed training of the prior more fleshed out?

Reply:
yup, we can always leave this for later!